### PR TITLE
Convert two old JMock based tests in Spock specifications

### PR DESCRIPTION
**Pull request by ljacomet**
This is a follow up to #4304 where I touched one of these tests and did
not clean it up eagerly

### CODE REVIEW

File: subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        Throwable failure = new RuntimeException()
+        DefaultExceptionAnalyser analyser = analyser()
+
+        when:
+        Throwable transformed = analyser.transform(failure)
+
+        then:
+        transformed instanceof LocationAwareException
+
+        LocationAwareException gse = (LocationAwareException) transformed
```
Comment (melix) [inline subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy]: I guess you made casting and types explicit to get IDE support, but it's not necessary

File: subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        gse.getCause().is(failure)
```
Comment (melix) [inline subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy]: More idiomatic groovy: `gse.cause.is(failure)`

File: subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+        gse.getReportableCauses().isEmpty()
```
Comment (melix) [inline subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy]: more idiomatic Groovy: `gse.reportableCauses.empty`

File: subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+        Throwable cause = new ContextualException()
+        Throwable failure = new ContextualException(cause)
+        DefaultExceptionAnalyser analyser = analyser()
+
+        when:
+        Throwable transformedFailure = analyser.transform(failure)
+
+        then:
+        LocationAwareException gse = (LocationAwareException) transformedFailure
+        gse.getCause()is(failure)
```
Comment (melix) [inline subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy]: `gse.cause.is(failure)` would be more readable (lack of space or dot in your code)

File: subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+        DefaultExceptionAnalyser analyser = analyser()
+
+        when:
+        notifyAnalyser(analyser, source)
+        Throwable transformedFailure = analyser.transform(failure)
+
+        then:
+        LocationAwareException gse = (LocationAwareException) transformedFailure
+        gse.getSourceDisplayName() == source.getDisplayName()
+        gse.getLineNumber() == 7
```
Comment (melix) [inline subprojects/core/src/test/groovy/org/gradle/initialization/DefaultExceptionAnalyserTest.groovy]: `gse.lineNumber == 7` (you get the idea ;))

File: subprojects/core/src/test/groovy/org/gradle/initialization/DefaultSettingsTest.groovy [hunk 6]
```diff
@@ -33,176 +31,153 @@ import org.gradle.configuration.ScriptPluginFactory
...
+    ScriptHandler settingsScriptHandler = Mock(ScriptHandler)
+    DefaultPluginManager pluginManager = Mock(DefaultPluginManager)
+    FeaturePreviews previews = new FeaturePreviews()
+    DefaultSettings settings
+
+    def setup() {
+        fileResolver.resolve(_) >> { args -> args[0].canonicalFile }
+
+        def settingsServices = Mock(ServiceRegistry)
+        settingsServices.get((Type)FileResolver.class) >> fileResolver
```
Comment (melix) [inline subprojects/core/src/test/groovy/org/gradle/initialization/DefaultSettingsTest.groovy]: `settingsServices.get(FileResolver)` should work too

**Comment by ljacomet** [comment]
@melix Thanks for the hints on idiomatic Groovy - changes done

**Pull request merged**